src/workspaces/list/RenderedWorkspaces.tsx

```diff
@@ -0,0 +1,306 @@
+
+  const canView = canRead(accessLevel);
+  const canAccessWorkspace = () =>
+    !canView ? setUserActions({ requestingAccessWorkspaceId: workspaceId }) : undefined;
+
+  return (
+    <div style={styles.tableCellContainer}>
+      <div style={styles.tableCellContent}>
+        <Link
+          aria-haspopup={canView ? undefined : 'dialog'}
+          style={{
+            ...(canView ? {} : { color: colors.dark(0.8), fontStyle: 'italic' }),
+            fontWeight: 600,
+            fontSize: 16,
+            ...Style.noWrapEllipsis,
+          }}
+          href={canView ? getLink('workspace-dashboard', { namespace, name }) : undefined}
+          onClick={() => {
+            canAccessWorkspace();
+            !!canView && Ajax().Metrics.captureEvent(Events.workspaceOpenFromList, extractWorkspaceDetails(workspace));
+          }}
+          tooltip={
+            !canView && 'You do not have access to this workspace. Select the workspace to learn about gaining access.'
+          }
+          tooltipSide='right'
+          disabled={workspace.state !== 'Ready'}
+        >
+          {name}
+        </Link>
+      </div>
+      <WorkspaceStateCell {...props} />
+    </div>
+  );
+};
+
+const LastModifiedCell = (props: CellProps): ReactNode => {
+  const {
+    workspace: { lastModified },
+  } = props.workspace;
+  return (
+    <div style={styles.tableCellContainer}>
+      <div style={styles.tableCellContent}>
+        <TooltipTrigger content={Utils.makeCompleteDate(lastModified)}>
+          <div>{Utils.makeStandardDate(lastModified)}</div>
+        </TooltipTrigger>
+      </div>
+    </div>
+  );
+};
+
+const CreatedByCell = (props: CellProps): ReactNode => (
+  <div style={styles.tableCellContainer}>
+    <div style={styles.tableCellContent}>
+      <span style={Style.noWrapEllipsis}>{props.workspace.workspace.createdBy}</span>
+    </div>
+  </div>
+);
+
+const AccessLevelCell = (props: CellProps): ReactNode => (
+  <div style={styles.tableCellContainer}>
+    <div style={styles.tableCellContent}>{Utils.normalizeLabel(props.workspace.accessLevel)}</div>
+  </div>
+);
+
+const CloudPlatformCell = (props: CellProps): ReactNode => {
+  return (
+    <div style={{ ...styles.tableCellContainer, paddingRight: 0 }}>
+      <div style={styles.tableCellContent}>
+        <CloudProviderIcon cloudProvider={getCloudProviderFromWorkspace(props.workspace)} />
+      </div>
+    </div>
+  );
+};
+
+interface ActionsCellProps extends CellProps {
+  workspaces: Workspace[];
+}
+
+const ActionsCell = (props: ActionsCellProps): ReactNode => {
+  const {
+    accessLevel,
+    workspace: { workspaceId, namespace, name, state },
+  } = props.workspace;
+  const { setUserActions } = useContext(WorkspaceUserActionsContext);
+
+  if (!canRead(accessLevel)) {
+    // No menu shown if user does not have read access.
+    return <div className='sr-only'>You do not have permission to perform actions on this workspace.</div>;
+  }
+  if (state === 'Deleted') {
```

Comment:
Mucking with that does not seems like a great idea right now.

Not showing the menu when the workspace has been deleted makes sense, because the workspace literally no longer exists, and the row is just a placeholder so the grid doesn't jump around.

But for the rest, there are still potentially valid actions, like share or leave.